tests/unit/btree_test.c

```diff
@@ -414,7 +414,7 @@ leaf_split_tests(btree_config    *cfg,
       slice key = slice_create(1, &i);
 
       bool success = btree_leaf_incorporate_tuple(
-         cfg, scratch, hdr, key, bigger_msg, &spec, &generation);
+         cfg, hid, hdr, key, bigger_msg, &spec, &generation);
```

Comment:
The reasons behind this fix has been aired out in internal [slack thread, here](https://vmware.slack.com/archives/G01MEDQ05MX/p1665524515654859?thread_ts=1665524450.401499&cid=G01MEDQ05MX).

Quoting from that response, here is my reasoning for this fix:

----
scratch is btree_scratch   *scratch declared as on-stack scratch buffers in CTEST_DATA().

On L445, passing down scratch as hid is benign now, as nothing is done to free this memory in shmem code-path.

By default, w/o using shared memory, in platform_free_from_heap() we will naturally fall through to free().

But in that function, due to above usage, because hid is non-NULL, we end up going to splinter_shm_free() which does nothing currently.

I just had enabled tracing in platform_free() to see who's calling it, and found that  `writable_buffer_deinit()` -> `platform_free()` trips that trace message that something is being freed from shared-memory, even though test was run without use of --use-shmem flag.

Walking back ... I found this location on L445 which seems problematic. (edited)